micrometer-core/src/test/java/io/micrometer/core/instrument/simple/SimpleMeterRegistryTest.java

```diff
@@ -101,6 +112,77 @@ public void newFunctionCounterWhenCountingModeIsStepShouldReturnStepFunctionCoun
         assertThat(functionCounter).isInstanceOf(StepFunctionCounter.class);
     }
 
+    @Test
+    @Disabled("Used only to show toString example output and does not test anything")
+    public void toStringOutputExample() throws InterruptedException {
+        final long testStart = System.nanoTime();
+        final Random random = new Random();
+        final SimpleMeterRegistry registry = new SimpleMeterRegistry();
+
+        final Counter requestSizeCounter = Counter.builder("request.size")
+                .baseUnit("bytes")
+                .register(registry);
+
+        final DistributionSummary responseSizeSummary = DistributionSummary.builder("response.size")
+                .baseUnit("bytes")
+                .publishPercentileHistogram()
+                .publishPercentiles(0.9999, 0.99, 0.95, 0.90, 0.75)
+                .sla(250, 500, 1000, 2000, 4000, 6000)
+                .register(registry);
+
+        final Timer responseTimer = Timer.builder("response.time")
+                .tag("action", "create")
+                .publishPercentiles(0.999, 0.99, 0.95, 0.90, 0.75)
+                .sla(Duration.ofMillis(20), Duration.ofMillis(50), Duration.ofMillis(100))
+                .publishPercentileHistogram()
+                .register(registry);
+
+        final AtomicLong testRunTime = new AtomicLong();
+        TimeGauge
+                .builder("test.run.time", testRunTime, TimeUnit.NANOSECONDS, AtomicLong::doubleValue)
+                .register(registry);
+
+        FunctionTimer
+                .builder("cache.latency", new Object(), o -> 10, o -> 300, TimeUnit.MILLISECONDS)
+                .register(registry);
+
+        FunctionCounter
+                .builder("cache.evictions", new Object(), o -> 10)
+                .baseUnit("beans")
+                .register(registry);
+
+        final LongTaskTimer taskTimer = LongTaskTimer
+                .builder("test.loop.timer")
+                .register(registry);
+
+        Meter.builder("default.meter", Type.GAUGE,
+                Arrays.asList(
+                        new Measurement(() -> 7.7, Statistic.VALUE),
+                        new Measurement(() -> 8.8, Statistic.VALUE)
+                ))
+                .baseUnit("beans")
+                .register(registry);
+
+        final LongTaskTimer.Sample loopSample = taskTimer.start();
+
+        for (int i = 0; i < 100 + random.nextInt(100); i++) {
+
+            final Sample testSample = Timer.start();
+            Thread.sleep(random.nextInt(100));
+            testSample.stop(responseTimer);
+
+            requestSizeCounter.increment(random.nextInt(300));
+
+            responseSizeSummary.record(random.nextInt(6000));
+        }
+
+        loopSample.stop();
+
+        testRunTime.set(System.nanoTime() - testStart);
+
+        System.out.println(registry.toString());
```

Comment:
Rather than printing it, it would be better to make an assertion that the result of `toString` matches a constant String here. Then, the result of the method will be tested and reviewers can see what the toString produces in this case.

Reply:
Hey, I added example output as a comment, you can see it's pretty huge, not sure how reliable that test would be, are you sure you want to assert on it?
Also now it is using random values, I would need to replace it with repeatable values.

Reply:
Asserting against the whole thing would be OK, assuming we can ensure the output is repeatable since we would be able to see the entirety.

Switch out the random values for actual concrete values should be sufficient.

Reply:
Done